library/Vanilla/Models/AddonModel.php

```diff
@@ -190,7 +190,9 @@ private function runSetup(Addon $addon) {
         // Look for a setup method.
         $called = $this->callPluginMethod($addon, 'setup');
 
-        if (($structure = $addon->getSpecial('structure')) && (!$called || !in_array($structure, get_included_files()))) {
+        // @TODO This if is a kludge because Vanilla's core applications are inconsistent.
+        // Once the InstallModel is in use this code can be cleaned up by manual structure inclusion in addons.
+        if (($structure = $addon->getSpecial('structure')) && (!$called || !in_array($addon->path($structure, Addon::PATH_FULL), get_included_files())) || $addon->getKey() === 'dashboard') {
```

Comment:
Note the comment I added. I realize this is a kludge right now, but this class is currently only being used in unit tests. As soon as its the one used to install Vanilla we can fix the offending inconsistent classes.